src/server/conn.rs

```diff
@@ -160,18 +166,33 @@ impl Conn {
             self.res.pop_front();
         }
 
-        // no data for writing, remove writable.
+        // no data for writing, remove writable
         self.interest.remove(EventSet::writable());
-        self.reregister(event_loop)
+        try!(self.reregister(event_loop));
+
+        self.on_write_complete(true);
```

Comment:
How can we ensure that the callback is called when the corresponding message rather than another one is written?